### PR TITLE
ENT-3337 | Create and Assign a new code upon enterprise unenrollment within the refund period

### PR DESCRIPTION
**Pull request by HammadAhmadWaqas**
this PR contains a new API that will be called from the LMS (with order number as param) to create a new order.
That order should be visible in the admin portal under the relevant filter if created successfully.

### CODE REVIEW

File: ecommerce/extensions/api/serializers.py [hunk 1]
```diff
@@ -1516,6 +1520,118 @@ def _trigger_email_sending_task(self, subject, greeting, closing, assigned_offer
             )
 
 
+class RefundedOrderCreateVoucherSerializer(serializers.Serializer):  # pylint: disable=abstract-method
+    """
+        Creates and assigns new coupon voucher to the requested order user.
```
Comment (irfanuddinahmad) [inline ecommerce/extensions/api/serializers.py]: Creates and assigns new coupon voucher to the user associated with the order.

File: ecommerce/extensions/api/serializers.py [hunk 2]
```diff
@@ -1516,6 +1520,118 @@ def _trigger_email_sending_task(self, subject, greeting, closing, assigned_offer
...
+    """
+    order = serializers.CharField(required=True)
+    code = serializers.CharField(read_only=True)
+
+    def validate_order(self, order):
+        """ Verify order number and returns order obj """
```
Comment (irfanuddinahmad) [inline ecommerce/extensions/api/serializers.py]: Verify the order number and return the order object

File: ecommerce/extensions/api/serializers.py [hunk 3]
```diff
@@ -1516,6 +1520,118 @@ def _trigger_email_sending_task(self, subject, greeting, closing, assigned_offer
...
+            name=coupon_product.title
+        )
+
+        attach_vouchers_to_coupon_product(
+            coupon_product,
+            vouchers,
+            note,
+        )
+
+        subject = 'Your New edX Access Code'
```
Comment (irfanuddinahmad) [inline ecommerce/extensions/api/serializers.py]: Move these to the constants file

File: ecommerce/extensions/api/serializers.py [hunk 4]
```diff
@@ -1516,6 +1520,118 @@ def _trigger_email_sending_task(self, subject, greeting, closing, assigned_offer
...
+                   'window. As part of our policy, we are giving you a new code to use. You can use this to enroll ' \
+                   'in any course that was available for you to enroll in with the original code.'
+        closing = 'We hope you find a course that meets your learning needs! For any questions, reach out to your ' \
+                  'Learning Manager at your organization.'
+        new_code = vouchers[0].code
+        serializer = CouponCodeAssignmentSerializer(
+            data={'codes': [new_code], 'emails': user_emails},
+            context={'coupon': coupon_product, 'subject': subject, 'greeting': greeting, 'closing': closing}
+        )
+        if serializer.is_valid():
```
Comment (irfanuddinahmad) [inline ecommerce/extensions/api/serializers.py:1590]: Handle the error case also

**Review by irfanuddinahmad — APPROVED** [review]
LGTM!

**Review by saleem-latif — APPROVED** [review]

**Review by moconnell1453 — APPROVED** [review]

**Pull request merged**